File: runtime/src/de/fatox/meta/api/dao/RenderBufferData.java
package de.fatox.meta.api.dao;

/**
 * Created by Frotty on 08.04.2017.
 */
public class RenderBufferData {
    public enum IN {
        GEOMETRY,
        FULLSCREEN
    }
    public IN inType = IN.GEOMETRY;
    public boolean hasDpeth = false;
    public String metaShaderPath;

    public RenderBufferData() {
    }

    public RenderBufferData(String metaShaderPath) {
        this.metaShaderPath = metaShaderPath;
    }

}

File: runtime/src/de/fatox/meta/api/dao/RenderBufferData.kt
package de.fatox.meta.api.dao

/**
 * Holds the data of a single Buffer inside a shader composition
 */
data class RenderBufferData(var metaShaderPath: String = "") {
    /** The input data type. GEOMETRY refers to all entities, FULLSCREEN to a fullscreen Quad (for post effects) */
    var inType = IN.GEOMETRY
    /** Whether or not this Buffer has a depth buffer */
    var hasDepth = true

    enum class IN {
        GEOMETRY,
        FULLSCREEN
    }

}
